src/rhash/hash.c

```diff
@@ -1143,6 +1143,76 @@ static int rc_hash_nintendo_ds(char hash[33], const char* path)
   return rc_hash_finalize(&md5, hash);
 }
 
+static uint32_t rc_read_bigendian_u32_value(void* file_handle)
+{
+  char le_buffer[4];
+  char be_buffer[4];
+  rc_file_read(file_handle, le_buffer, 4);
+  for (int ix = 0; ix < 4; ix++)
+  {
+    be_buffer[3 - ix] = le_buffer[ix];
+  }
+  return *((uint32_t*)be_buffer);
+}
+
+static int rc_hash_gamecube(char hash[33], const char* path)
+{
+  md5_state_t md5;
+  void* file_handle;
+  const int BASE_HEADER_SIZE = 0x2440;
+
+  file_handle = rc_file_open(path);
```

Comment:
For the GC and Wii, you can make an assumption here regarding those factors. GC and Wii used special drives, they didn't support all the fancy formats you could find on CDs and such. They were limited to a single track, and did not support the extra metadata, so they were always 2048 sector ISOs. Dolphin goes under this assumption too.

Reply:
ISO is the MVP at the moment. I definitely wish to add RVZ support down the road (especially since Dolphin devs so heavily emphasize that format) but there's still questions about whether the decompression happens inside rcheevos or externally in whichever emulator is calling it.